libs/core/src/lib/radio/radio-button/radio-button.component.ts

```diff
@@ -0,0 +1,213 @@
+    /** @hidden */
+    class: string;
+
+    /** @hidden */
+    @ViewChild('inputElement', { static: false })
+    inputElement: ElementRef;
+
+    /** @hidden */
+    actualValue: any;
+
+    private _compact: boolean = false;
+
+    /** Whether to apply compact mode to the radio button.
+     * Value: true or false
+     * By default field is set to false
+     */
+    @Input()
+    set compact(isCompact: boolean) {
+        this._compact = isCompact;
+        this.buildComponentCssClass();
+    }
+
+    private _state: stateType = 'default';
+    /** The field to set state of radio button using:
+     * 'valid' | 'invalid' | 'warning' | 'default' | 'information'
+     * by default value is set to 'default'
+     */
+    @Input()
+    set state(newState: stateType) {
+        this._state = newState;
+        this.buildComponentCssClass();
+    }
+
+    private _disabled: boolean = false;
+    /** The field is used to tell if radio button should be disabled
+     * Value: true or false
+     * by default disabled state is set to false
+     */
+    @Input()
+    set disabled(isDisabled: boolean) {
+        this._disabled = isDisabled;
+        this.buildComponentCssClass();
+    }
+
+    /** @hidden */
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    /** The field should be only used with reactive forms
+     * Its purpose is to pass a current selected value from froumGroup
+     * The field is mandatory when working with reactive forms
+     */
+    @Input()
+    set selectedValue(v: any) {
+        this.actualValue = v;
+    }
+
+    private _name: string;
+    /** The name of the radio button
+     * The field is mandatory
+     */
+    @Input()
+    set name(v: string) {
+        this._name = v;
+    }
+
+    /** @hidden */
+    get name(): string {
+        return this._name;
+    }
+
+    get checked(): boolean {
+        if (this.value === undefined) {
+            return false;
+        }
+        return this.actualValue === this.value;
+    }
+
+    /**
+     * Set uniqueId to a radio button
+     */
+    @Input()
+    id: string = `radio-id-${uniqueId++}`;
+
+    /** Value field stores information about holding value by radio button
+     * The field is mandatory
+     */
+    @Input()
+    value: any;
+
+    // ControlValueAccessor implementation
+    /** @hidden */
+    onChange: any = () => {};
+
+    /** @hidden */
+    onTouched: any = () => {};
+
+    /** @hidden */
+    registerOnChange(fn: (selected: any) => { void }): void {
+        this.onChange = fn;
+    }
+
+    /** @hidden */
+    registerOnTouched(fn: any): void {
+        this.onTouched = fn;
+    }
+
+    /** @hidden */
+    setDisabledState(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+        this.changeDetectionRef.detectChanges();
+    }
+
+    /** @hidden */
+    writeValue(value: any): void {
+        this.valueChange(value);
```

Comment:
Why writing new value calls `valueChange` and then inside `valueChange` `this.onChange(value);`?
Is that correct flow? It looks like custom control updates FormControl with value just received from FormControl.

Reply:
I tested it and it is required otherwise this happens 
![image](https://user-images.githubusercontent.com/10849982/72148984-662a7b00-33a2-11ea-814f-95f6d90110e8.png)